specs/destination.go

```diff
@@ -11,15 +11,18 @@ import (
 
 type WriteMode int
 
+type MigrateMode int
+
 type Destination struct {
-	Name           string    `json:"name,omitempty"`
-	Version        string    `json:"version,omitempty"`
-	Path           string    `json:"path,omitempty"`
-	Registry       Registry  `json:"registry,omitempty"`
-	WriteMode      WriteMode `json:"write_mode,omitempty"`
-	BatchSize      int       `json:"batch_size,omitempty"`
-	BatchSizeBytes int       `json:"batch_size_bytes,omitempty"`
-	Spec           any       `json:"spec,omitempty"`
+	Name           string      `json:"name,omitempty"`
+	Version        string      `json:"version,omitempty"`
+	Path           string      `json:"path,omitempty"`
+	Registry       Registry    `json:"registry,omitempty"`
+	WriteMode      WriteMode   `json:"write_mode,omitempty"`
+	MigrateMode    MigrateMode `json:"migrate_mode,omitempty"`
```

Comment:
This can also be a boolean, just found it easier to communicate the purpose using strings